File: QID-1616-MVAregbank/MVAregbank.R

# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

# load data
x  = read.table("bank2.dat")
X4 = x[1:100, 4]
X5 = x[1:100, 5]

b = lm(X5 ~ X4)$coefficients  # regression of (X5) on (X4) price
Y = b[1] + b[2] * X4  # regression line

# plot
plot(X4, X5, xlab = "Lower inner frame (X4), genuine", ylab = "Upper inner frame (X5), genuine", 
    xlim = c(7, 10.5), ylim = c(7.5, 12), frame = TRUE, axes = FALSE)
lines(X4, Y, col = "red3", lwd = 2)
axis(side = 2, seq(7.5, 12.5, 1), seq(7.5, 12.5, 1))
axis(side = 1, seq(7, 12, 1), seq(7, 12, 1))
title("Swiss bank notes")

File: QID-1616-MVAregbank/MVAregbank.py
# works on pandas 1.5.2, numpy 1.23.5 and matplotlib 3.6.3
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

x = pd.read_csv("bank2.dat", sep = "\s+", header=None)

x1 = x.iloc[:100,3]
y = x.iloc[:100,4]

a, b = np.polyfit(x1,y,1)
xfit = np.linspace(x1.min(),x1.max(),400)
yfit = b + a*xfit

fig, ax = plt.subplots(figsize = (10, 10))
ax.scatter(x1, y, c = "black")
ax.plot(xfit,yfit, c = "r")
plt.xlabel("Lower inner frame(X4), genuine", fontsize = 20)
plt.ylabel("Upper inner frame(X5), genuine", fontsize = 20)
plt.title("Swiss bank notes", fontsize = 25, weight="bold")

plt.show()
